src/userspace/exec_monitor.cpp

```diff
@@ -16,26 +16,47 @@
 #include <unistd.h>
 #include <bpf/bpf.h>
 #include <iostream>
+#include <functional>
 #include "exec_monitor.skel.h"
 #include "process_info.hpp"
+#include "process_info.pb.h"
 #include "exec_monitor.hpp"
 
+std::fstream file;
+ExportFormat export_format;
 static int process_sample(void *ctx, void *data, size_t len)
 {
-	if(len < sizeof(struct process_info)) {
+	if(len < sizeof(struct process_info))
 		return -1;
-	}
 
 	struct process_info *s = (process_info*)data;
-	printf("%d\t%d\t%d\t%s\n", s->ppid, s->pid, s->tgid, s->name);
+	ExecMonitor *exec_monitor = exec_monitor->get_instance();
+	exec_monitor->export_data(s);
 	return 0;
 }
 
-ExecMonitor::ExecMonitor(Config config)
+ExecMonitor *ExecMonitor::instance = 0;
+ExecMonitor::ExecMonitor(Config *config)
+{
+	n_proc = config->n_proc;
```

Comment:
I would prefer accessing everything with `Config::get_instance()->n_proc`, this would make it clearer what comes from the config and what comes from something else. Also it keeps all the config in the Config class, which is what you would expect from a class that is only here to encapsulate data.